File: advent-of-code/2023/day1/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"unicode"
)

func extractTwoDigitNumber(line string) int {
	var frontNumber int = -1
	for _, c := range line {
		if unicode.IsDigit(c) {
			frontNumber = int(c - '0')
			break
		}
	}

	var backNumber int = -1
	for i := len(line) - 1; i >= 0; i-- {
		if unicode.IsDigit(rune(line[i])) {
			backNumber = int(line[i] - '0')
			break
		}
	}

	if frontNumber != -1 && backNumber != -1 {
		return frontNumber*10 + backNumber
	}
	return 0
}

func main() {
	file, err := os.Open("/home/abhinav/code/codebase/advent-of-code/2023/day1/input.txt")
	if err != nil {
		fmt.Println("Error opening file", err)
		return
	}
	defer file.Close()

	sum := 0
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		sum += extractTwoDigitNumber(line)
	}

	if err := scanner.Err(); err != nil {
		fmt.Println("Error reading file:", err)
		return
	}

	fmt.Printf("Total sum: %d\n", sum)
}

File: advent-of-code/2023/day1/day1.py
def extract_front_num(line):
    front_num = -1
    for i in line:
        if i.isdigit():
            front_num = int(i)
            return front_num


def extract_last_num(line):
    last_num = -1
    for i in reversed(line):
        if i.isdigit():
            last_num = int(i)
            return last_num


def main():
    sum = 0
    with open(
        "/home/abhinav/code/codebase/advent-of-code/2023/day1/input.txt", "r"
    ) as file:
        for line in file:
            f = extract_front_num(line.strip())
            l = extract_last_num(line.strip())

            if f != -1 and l != -1:
                sum += f * 10 + l
        print(sum)


if __name__ == "__main__":
    main()
